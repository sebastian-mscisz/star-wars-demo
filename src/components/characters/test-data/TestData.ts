import { Characters, Character, Species } from 'src/common';

export const TestCharacter: Character = {
    name: 'Character1',
    url: 'https://swapi.info/api/people/1',
    homeworld: 'https://swapi.info/api/planets/1',
    species: ['species/1'],
    vehicles: ['https://swapi.info/api/vehicles/1'],
};

export const TestCharacters: Characters = [
    TestCharacter,
    {
        name: 'Character2',
        url: 'https://swapi.info/api/people/2',
        vehicles: ['https://swapi.info/api/vehicles/1', 'https://swapi.info/api/vehicles/2'],
        homeworld: 'https://swapi.info/api/planets/2',
        species: ['species/2'],
    },
    {
        name: 'Character3',
        url: 'https://swapi.info/api/people/3',
        vehicles: [],
        homeworld: 'https://swapi.info/api/planets/3',
        species: ['species/3'],
    },
    {
        name: 'Character4',
        url: 'https://swapi.info/api/people/4',
        vehicles: ['https://swapi.info/api/vehicles/3'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/4'],
    },
    {
        name: 'Character5',
        url: 'https://swapi.info/api/people/5',
        vehicles: ['https://swapi.info/api/vehicles/4', 'https://swapi.info/api/vehicles/5'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/5'],
    },
];

export const TestSpecies: Species = [
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species3',
        url: 'species/3',
    },
    {
        name: 'Species4',
        url: 'species/4',
    },
    {
        name: 'Species5',
        url: 'species/5',
    },
];
