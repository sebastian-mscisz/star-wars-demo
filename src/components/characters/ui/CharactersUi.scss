@import 'src/variables';

.sw-characters {
    --gap: 20px;
    --columns: 7;

    @media screen and (max-width: $xlarge-bp) {
        --columns: 6;
    }

    @media screen and (max-width: $large-bp) {
        --columns: 5;
    }

    @media screen and (max-width: $medium-bp) {
        --columns: 4;
    }

    @media screen and (max-width: $small-bp) {
        --columns: 3;
    }

    display: flex;
    flex-wrap: wrap;
    gap: var(--gap);
    padding: 20px;
}
