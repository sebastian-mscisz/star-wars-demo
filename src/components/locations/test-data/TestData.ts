import { Locations, Location } from 'src/common';

export const TestLocation: Location = {
    name: 'Location1',
    url: 'https://swapi.info/api/planets/1',
    residents: ['https://swapi.info/api/people/1'],
    population: '1000',
};

export const TestLocations: Locations = [
    TestLocation,
    {
        name: 'Location2',
        url: 'https://swapi.info/api/planets/2',
        residents: ['https://swapi.info/api/people/2'],
        population: '5343',
    },
    {
        name: 'Location3',
        url: 'https://swapi.info/api/planets/3',
        residents: ['https://swapi.info/api/people/3'],
        population: '13333',
    },
    {
        name: 'Location4',
        url: 'https://swapi.info/api/planets/4',
        residents: ['https://swapi.info/api/people/4', 'https://swapi.info/api/people/5'],
        population: '9999',
    },
    {
        name: 'Location5',
        url: 'https://swapi.info/api/planets/5',
        residents: [],
        population: '999',
    },
];
