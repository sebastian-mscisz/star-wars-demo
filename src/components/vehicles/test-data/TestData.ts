import { Vehicles, Vehicle } from 'src/common';

export const TestVehicle: Vehicle = {
    name: 'Vehicle1',
    url: 'https://swapi.info/api/vehicles/1',
    vehicle_class: 'Air speeder',
    pilots: ['character/1', 'character/2'],
};

export const TestVehicles: Vehicles = [
    TestVehicle,
    {
        name: 'Vehicle2',
        url: 'https://swapi.info/api/vehicles/2',
        pilots: ['character/2'],
        vehicle_class: 'Land speeder',
    },
    {
        name: 'Vehicle3',
        url: 'https://swapi.info/api/vehicles/3',
        pilots: ['character/4'],
        vehicle_class: 'Water speeder',
    },
    {
        name: 'Vehicle4',
        url: 'https://swapi.info/api/vehicles/4',
        pilots: ['character/5'],
        vehicle_class: 'Fire speeder',
    },
    {
        name: 'Vehicle5',
        url: 'https://swapi.info/api/vehicles/5',
        pilots: ['character/5'],
        vehicle_class: 'Air speeder',
    },
];
