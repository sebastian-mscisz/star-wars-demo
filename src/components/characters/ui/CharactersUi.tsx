import { Characters, charactersPagePath, sortAZ } from 'src/common';
import { CharactersCard, ItemsWithPagination } from 'src/components';
import './CharactersUi.scss';
import { useParams } from 'react-router-dom';

type CharactersUiProps = {
    characters?: Characters;
};

const itemsPerPage = 21;

export const CharactersUi: React.FC<CharactersUiProps> = ({ characters }) => {
    const pageId = useParams();
    const currentPageId = Number(Object.values(pageId)[0]) || 1;

    if (!characters) return null;

    const sortedCharacters = [...characters].sort((a, b) => sortAZ(a.name || '', b.name || ''));

    return (
        <ItemsWithPagination
            items={sortedCharacters}
            itemsPerPage={itemsPerPage}
            currentPageId={currentPageId}
            navPagePath={charactersPagePath()}
            className={'characters'}
            CardComponent={CharactersCard}
        />
    );
};
