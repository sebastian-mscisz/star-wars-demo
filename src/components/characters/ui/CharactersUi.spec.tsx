import { BrowserRouter } from 'react-router-dom';
import { CharactersUi } from './CharactersUi';
import { render } from '@testing-library/react';
import { TestCharacters } from '../test-data';

describe('CharactersUi', () => {
    it('should render', () => {
        const { asFragment } = render(<CharactersUi characters={TestCharacters} />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
