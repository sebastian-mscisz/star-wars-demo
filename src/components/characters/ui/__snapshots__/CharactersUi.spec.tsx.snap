// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersUi should render 1`] = `
<DocumentFragment>
  <div
    class="sw-pagination-section"
    data-testid="pagination-section"
  >
    <div
      class="sw-pagination-button sw-pagination-button--hidden"
      data-testid="left-pagination-button"
    >
      ←
    </div>
    <div
      class="sw-pagination-button sw-pagination-button--hidden"
      data-testid="right-pagination-button"
    >
      →
    </div>
  </div>
  <div
    class="sw-characters"
  >
    <div
      class="sw-characters-card"
    >
      <div
        class="sw-characters-card__tile"
      >
        <a
          class="sw-characters-card__tile__link"
          href="/character/1"
        >
          <div
            class="sw-characters-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars character"
              loading="lazy"
              src="./characters/1.jpg"
            />
          </div>
          <h5
            class="sw-characters-card__tile__link__text"
          >
            Character1
          </h5>
        </a>
      </div>
    </div>
    <div
      class="sw-characters-card"
    >
      <div
        class="sw-characters-card__tile"
      >
        <a
          class="sw-characters-card__tile__link"
          href="/character/2"
        >
          <div
            class="sw-characters-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars character"
              loading="lazy"
              src="./characters/2.jpg"
            />
          </div>
          <h5
            class="sw-characters-card__tile__link__text"
          >
            Character2
          </h5>
        </a>
      </div>
    </div>
    <div
      class="sw-characters-card"
    >
      <div
        class="sw-characters-card__tile"
      >
        <a
          class="sw-characters-card__tile__link"
          href="/character/3"
        >
          <div
            class="sw-characters-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars character"
              loading="lazy"
              src="./characters/3.jpg"
            />
          </div>
          <h5
            class="sw-characters-card__tile__link__text"
          >
            Character3
          </h5>
        </a>
      </div>
    </div>
    <div
      class="sw-characters-card"
    >
      <div
        class="sw-characters-card__tile"
      >
        <a
          class="sw-characters-card__tile__link"
          href="/character/4"
        >
          <div
            class="sw-characters-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars character"
              loading="lazy"
              src="./characters/4.jpg"
            />
          </div>
          <h5
            class="sw-characters-card__tile__link__text"
          >
            Character4
          </h5>
        </a>
      </div>
    </div>
    <div
      class="sw-characters-card"
    >
      <div
        class="sw-characters-card__tile"
      >
        <a
          class="sw-characters-card__tile__link"
          href="/character/5"
        >
          <div
            class="sw-characters-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars character"
              loading="lazy"
              src="./characters/5.jpg"
            />
          </div>
          <h5
            class="sw-characters-card__tile__link__text"
          >
            Character5
          </h5>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
